File: src/NetFxCSharp/Facesso.Interfaces/Kannegiesser/KannegiesserProductionDataImportClasses_Worker.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Data.OleDb;
using System.Diagnostics;
using System.Linq;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace Facesso.Interfaces
{
    public partial class KannegiesserProductionDataImportTaskElement
    {
        private void retrieveDataForDate(System.DateTime ProductionDate)
        {
            if (PathToDeviceData == null)
            {
                return;
            }

            //Quelldaten für den Tag abrufen
            OleDbConnection locConnection = new OleDbConnection(ConnectionString);
            DataTable locSourceData = new DataTable();
            using (locConnection)
            {
                OleDbDataAdapter locAdapter = new OleDbDataAdapter("SELECT * FROM PROTOKOL WHERE DATUM=" + ProductionDate.ToString("\\#MM\\/dd\\/yyyy\\#") + " AND TYP=0 AND ARTNR>0", locConnection);
                //Falls das eine Exception wirft, dann den BDE (Paradox-Treiber) installieren.
                //Der ist nicht mehr erhältlich, heißt bde.exe ist im Install-Verzeichnis auf
                //dem Server, ca. 5 MByte groß.
                int locIntBack = locAdapter.Fill(locSourceData);
            }

            //Zieldatentabellenstruktur einrichten
            myCurrFacData = new ProductionDataTable();
            //Daten umschaufeln und dabei neu berechnen
            foreach (DataRow locDataRow in locSourceData.Rows)
            {
                ProductionDataRow locFacRow = default(ProductionDataRow);
                locFacRow = myCurrFacData.NewProductionDataRow();
                {
                    var __with0 = locFacRow;
                    DateTime locStartTime = ProductionDate.AddSeconds(System.Convert.ToInt32(locDataRow["STARTZEIT"]));
                    __with0.StartTime = locStartTime;
                    __with0.EndTime = __with0.StartTime.AddSeconds(System.Convert.ToInt32(locDataRow["DAUER"]));
                    __with0.ProgramNo = System.Convert.ToInt32(locDataRow["ARTNR"]);
                    __with0.TotalAmount = System.Convert.ToDouble(locDataRow["ZAEHLER"]);
                    ShiftTimeSpan locShiftSpan = this.ForWorkgroup.TimeSettingDetails.FindShiftForStartTime(ProductionDate, locStartTime);
                    if (locShiftSpan != null)
                    {
                        __with0.Shift = this.ForWorkgroup.TimeSettingDetails.FindShiftForStartTime(ProductionDate, locStartTime).ShiftNo;
                    }
                    else
                    {
                        locShiftSpan = this.ForWorkgroup.TimeSettingDetails.FindShiftForStartTime(ProductionDate, locStartTime);
                    }
                }

                myCurrFacData.AddProductionDataRow(locFacRow);
            }
        }
    }
}

File: src/NetFxCSharp/Facesso.Interfaces/Kannegiesser/KannegiesserProductionDataImportClasses_Worker.vb
Imports System.Data.OleDb

Partial Class KannegiesserProductionDataImportTaskElement

    Private Sub retrieveDataForDate(ByVal ProductionDate As Date)

        If PathToDeviceData Is Nothing Then
            Return
        End If

        'Quelldaten f¸r den Tag abrufen
        Dim locConnection As New OleDbConnection(ConnectionString)

        Dim locSourceData As new DataTable
        Using locConnection
            Dim locAdapter As New OleDbDataAdapter("SELECT * FROM PROTOKOL WHERE DATUM=" & _
                    ProductionDate.ToString("\#MM\/dd\/yyyy\#") & _
                    " AND TYP=0 AND ARTNR>0", locConnection)
            'Falls das eine Exception wirft, dann den BDE (Paradox-Treiber) installieren.
            'Der ist nicht mehr erh‰ltlich, heiﬂt bde.exe ist im Install-Verzeichnis auf
            'dem Server, ca. 5 MByte groﬂ.
            Dim locIntBack As Integer = locAdapter.Fill(locSourceData)
        End Using

        'Zieldatentabellenstruktur einrichten
        myCurrFacData = New ProductionDataTable

        'Daten umschaufeln und dabei neu berechnen
        For Each locDataRow As DataRow In locSourceData.Rows
            Dim locFacRow As ProductionDataRow
            locFacRow = myCurrFacData.NewProductionDataRow()
            With locFacRow
                Dim locStartTime As DateTime = ProductionDate.AddSeconds(CInt(locDataRow("STARTZEIT")))
                .StartTime = locStartTime
                .EndTime = .StartTime.AddSeconds(CInt(locDataRow("DAUER")))
                .ProgramNo = CInt(locDataRow("ARTNR"))
                .TotalAmount = CDbl(locDataRow("ZAEHLER"))
                Dim locShiftSpan As ShiftTimeSpan = Me.ForWorkgroup.TimeSettingDetails.FindShiftForStartTime(ProductionDate, locStartTime)
                If locShiftSpan IsNot Nothing Then
                    .Shift = Me.ForWorkgroup.TimeSettingDetails.FindShiftForStartTime(ProductionDate, locStartTime).ShiftNo
                Else
                    locShiftSpan = Me.ForWorkgroup.TimeSettingDetails.FindShiftForStartTime(ProductionDate, locStartTime)
                End If
            End With
            myCurrFacData.AddProductionDataRow(locFacRow)
        Next
    End Sub
End Class
